opensrp-chw-core/src/main/java/org/smartregister/chw/core/interactor/CoreChildProfileInteractor.java

```diff
@@ -482,16 +486,24 @@ public void createCarePlanEvent(@NotNull Context context, @NotNull String encode
                 // getting thinkMD id from encoded fhir bundle
                 String thinkMdId = ThinkMDLibrary.getInstance().getThinkMDPatientId(encodedBundle);
                 // getting the baseEntityId mapped to thinkMD
-                String baseEntityId = getBaseEntityID(context.getResources().getString(R.string.thinkmd_identifier_type), thinkMdId);
+                String childBaseEntityId = getBaseEntityID(thinkMdId);
                 // creating the event to sync with server
-                if (baseEntityId != null) {
+                if (childBaseEntityId != null) {
                     Event carePlanEvent = ThinkMDLibrary.getInstance().createCarePlanEvent(encodedBundle,
                             getFormTag(getAllSharedPreferences()),
-                            baseEntityId);
+                            childBaseEntityId);
+                    updateLocalStorage(childBaseEntityId, "thinkmd_fhir_bundle", encodedBundle);
+
+                    for (Obs obs : carePlanEvent.getObs()) {
+                        if (StringUtils.isEmpty(obs.getFormSubmissionField())) continue;
```

Comment:
do we really need this statement since the one below checks for a specific value

Reply:
Yes we need this statement, we need to query the obs and update the local DB with the care plan date and care plan.